compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/InterfaceLowering.kt

```diff
@@ -100,6 +107,59 @@ private class InterfaceLowering(val context: JvmBackendContext) : IrElementTrans
             }
         }
     }
+
+    override fun visitReturn(expression: IrReturn): IrExpression {
+        val newFunction = removedFunctions[expression.returnTargetSymbol]?.owner
+        return super.visitReturn(
+            if (newFunction != null) {
+                with(expression) {
+                    IrReturnImpl(startOffset, endOffset, expression.type, newFunction.symbol, expression.value)
+                }
+            } else {
+                expression
+            }
+        )
+    }
+
+    override fun visitCall(expression: IrCall): IrExpression {
+        val newFunction = removedFunctions[expression.symbol]?.owner
+        return super.visitCall(
+            if (newFunction != null) {
+                with(expression) {
+                    if (this is IrCallWithShallowCopy) {
+                        shallowCopy(origin, symbol, superQualifierSymbol)
```

Comment:
Shouldn't this be `newFunction.symbol`?

Reply:
You are correct, good catch! Actually, thanks for catching this, "correcting" it actually surfaced some errors here. `IrCallWithShallowCopy` is implemented by 4 classes:
1. `IrBinaryPrimitiveImpl`
1. `IrNullaryPrimitiveImpl`
1. `IrGetterCallImpl`
1. `IrSetterCallImpl`

The first 2 will never be present in an interface and moved by `InterfaceLowering`. The latter 2 do exist for properties in interfaces, and will be moved to `DefaultImpls`. However, these will no longer have the dispatch receiver parameters so doing the shallow copy does not work. Simplifying the code to:
```
with(expression) {
    irCall(this, newFunction, receiversAsFirstArguments = true)
}
```
...works and we can now enable 4 more tests. :slightly_smiling_face:

Reply:
Maybe make it even shorter then? :)

```
irCall(expression, newFunction, receiversAsFirstArguments = true)
```

I also have doubts about the other uses of `with`...

Reply:
Done. I use `with` just so I don't have to do `expression.` for offsets, type, etc. That said, I missed a few spots and fixed those now.